lib/src/test/java/io/ably/lib/test/realtime/RealtimeMessageTest.java

```diff
@@ -986,4 +970,43 @@ public void opaque_message_extras() throws AblyException {
             }
         }
     }
+
+    /**
+     * Publish a message that contains extras of invalid format.
+     */
+    @Test
+    public void opaque_message_extras_fail() throws AblyException {
+        AblyRealtime ably = null;
+        try {
+            final JsonObject opaqueJson = new JsonObject();
+            opaqueJson.addProperty("Some Property", "Lorem Ipsum");
+            opaqueJson.addProperty("Some Truth", false);
+            opaqueJson.addProperty("Some Number", 321);
+
+            final MessageExtras extras = new MessageExtras(opaqueJson);
+            final Message message = new Message();
+            message.name = "The Test Message";
+            message.data = "Some Value";
+            message.extras = extras;
+
+            final ClientOptions clientOptions = createOptions(testVars.keys[0].keyStr);
+            ably = new AblyRealtime(clientOptions);
+
+            // create a channel and attach to it
+            final Channel channel = ably.channels.get(createChannelName("opaque_message_extras_fail"));
+            channel.attach();
+            (new ChannelWaiter(channel)).waitFor(ChannelState.attached);
+            assertEquals(ChannelState.attached, channel.state);
+
+            // publish and await success
```

Comment:
Perhaps clearer to adjust this comment:

```suggestion
            // publish and await expected failure (completion with error)
```